src/tools/rustbook/Cargo.toml

```diff
@@ -10,6 +10,6 @@ clap = "2.25.0"
 env_logger = "0.7.1"
 
 [dependencies.mdbook]
-version = "0.4.6"
+version = "^0.4"
```

Comment:
In cargo, if the version requirement does not have any operators, it is interpreted the same way as if we had specified caret requirements. Therefore, I think this PR does not actually change anything. See also: https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio

if you want to update the version of the mdbook used in CI, you have to update Cargo.lock.

Reply:
Maybe this could be solved by add cargo update in CI?